### PR TITLE
ARIA-138-Make-logging-more-informative

### PR DESCRIPTION
**Pull request by mxmrlv**

### CODE REVIEW

File: aria/cli/commands/__init__.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (mxmrlv) [inline aria/cli/commands/__init__.py]: remove

File: aria/cli/commands/executions.py [hunk 2]
```diff
@@ -141,13 +143,19 @@ def start(workflow_name,
 
     logger.info('Starting {0}execution. Press Ctrl+C cancel'.format('dry ' if dry else ''))
     execution_thread.start()
+
+    log_consumer = cli_logger.ModelLogConsumer(model_storage, workflow_runner.execution_id)
```
Comment (mxmrlv) [inline aria/cli/commands/executions.py]: Iterator

File: aria/cli/execution_logging.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+DEFAULT_FORMATTING = {
+    logger.NO_VERBOSE: {'main_msg': '{item.msg}'},
+    logger.LOW_VERBOSE: {
+        'main_msg': '{created_at} | {item.level[0]} | {item.msg}',
+        'created_at': '%H:%M:%S'
+    }
+}
+
+
+class load(object):
```
Comment (mxmrlv) [inline aria/cli/execution_logging.py]: remove class

File: aria/cli/execution_logging.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    def __repr__(self):
+        # Only NO_VERBOSE and LOW_VERBOSE are configurable formats. configuring
+        # the low verbose level should affect any higher level.
+        formats = self._formats[min(env.logging.verbosity_level, logger.LOW_VERBOSE)]
+
+        kwargs = dict(item=self._item)
+        if 'created_at' in formats:
+            kwargs['created_at'] = self._item.created_at.strftime(formats['created_at'])
+        if 'level' in formats:
```
Comment (mxmrlv) [inline aria/cli/execution_logging.py]: no need

File: aria/orchestrator/context/common.py [hunk 5]
```diff
@@ -38,43 +38,44 @@ class BaseContext(object):
...
 
     class PrefixedLogger(object):
-        def __init__(self, logger, prefix='', task_id=None):
-            self._logger = logger
-            self._prefix = prefix
+        def __init__(self, base_logger, task_id=None):
+            self._logger = base_logger
             self._task_id = task_id
 
         def __getattr__(self, item):
```
Comment (mxmrlv) [inline aria/orchestrator/context/common.py]: , attr...

File: aria/orchestrator/workflows/core/engine.py [hunk 6]
```diff
@@ -40,6 +41,7 @@ class Engine(logger.LoggerMixin):
 
     def __init__(self, executor, workflow_context, tasks_graph, **kwargs):
         super(Engine, self).__init__(**kwargs)
+        self.logger.addHandler(logging.NullHandler())
```
Comment (mxmrlv) [inline aria/orchestrator/workflows/core/engine.py]: look up NullHandlers in code and make sure it makes sense

File: aria/orchestrator/workflows/core/engine.py [hunk 7]
```diff
@@ -20,6 +20,7 @@
 import time
```
Comment (mxmrlv) [inline aria/orchestrator/workflows/core/engine.py]: visit dry.py and check output

File: aria/cli/commands/executions.py [hunk 8]
```diff
@@ -19,6 +19,8 @@
 from .. import table
 from .. import utils
 from ..core import aria
+from .. import logger as cli_logger
```
Comment (ran-z) [inline aria/cli/commands/executions.py]: move above the previous import

File: aria/cli/commands/logs.py [hunk 9]
```diff
@@ -12,13 +12,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-from .. import utils
-from ..core import aria
+from ..logger import ModelLogIterator
+from ..cli import aria
```
Comment (ran-z) [inline aria/cli/commands/logs.py]: this line is wrong

File: aria/cli/commands/executions.py [hunk 10]
```diff
@@ -161,11 +170,13 @@ def start(workflow_name,
         model_storage.execution.delete(execution)
 
 
-def _cancel_execution(workflow_runner, execution_thread, logger):
+def _cancel_execution(model_storage, workflow_runner, execution_thread, logger):
     logger.info('Cancelling execution. Press Ctrl+C again to force-cancel')
+    log_iterator = cli_logger.ModelLogIterator(model_storage, workflow_runner.execution_id)
```
Comment (mxmrlv) [inline aria/cli/commands/executions.py]: use existing

File: aria/cli/commands/executions.py [hunk 11]
```diff
@@ -141,12 +143,19 @@ def start(workflow_name,
...
     try:
         while execution_thread.is_alive():
-            # using join without a timeout blocks and ignores KeyboardInterrupt
+            execution_logging.log_list(log_iterator)
             execution_thread.join(1)
+
     except KeyboardInterrupt:
-        _cancel_execution(workflow_runner, execution_thread, logger)
+        _cancel_execution(model_storage, workflow_runner, execution_thread, logger)
```
Comment (mxmrlv) [inline aria/cli/commands/executions.py]: move model_storage later

File: aria/cli/commands/logs.py [hunk 12]
```diff
@@ -12,13 +12,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-from .. import utils
-from ..core import aria
+from ..logger import ModelLogIterator
+from .. core import aria
```
Comment (mxmrlv) [inline aria/cli/commands/logs.py]: ?

Comment (mxmrlv) [inline aria/cli/commands/logs.py]: should also be before logger

File: aria/orchestrator/workflows/core/engine.py [hunk 13]
```diff
@@ -40,6 +40,7 @@ class Engine(logger.LoggerMixin):
 
     def __init__(self, executor, workflow_context, tasks_graph, **kwargs):
         super(Engine, self).__init__(**kwargs)
+        self.logger.addHandler(logger.NullHandler())
```
Comment (mxmrlv) [inline aria/orchestrator/workflows/core/engine.py]: ???

**Pull request merged**